h/security/policy/bearer_token.py

```diff
@@ -1,12 +1,10 @@
-from pyramid.interfaces import ISecurityPolicy
 from pyramid.request import RequestLocalCache
-from zope.interface import implementer
 
 from h.security.identity import Identity
 from h.security.policy._identity_base import IdentityBasedPolicy
+from h.security.policy.helpers import is_api_request
 
 
-@implementer(ISecurityPolicy)
```

Comment:
This `@implementer(ISecurityPolicy)` was added to some of our security policies but not others. Security policies don't need to have this and the examples in the Pyramid docs don't have it. So I'm just removing it from all of them.

Reply:
To check my understanding: the purpose of `@implementer` is to add runtime metadata to a class to indicate that it implements an interface, and neither we nor Pyramid use that capability for these classes?

Reply:
Exactly, yes. It's normal both in Pyramid and in Python generally to have "informal" interfaces that the framework requires users to implement. Pyramid does define an [`ISecurityPolicy` interface](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/interfaces.html#pyramid.interfaces.ISecurityPolicy) and I believe it uses it under the hood but from the user's point of view it's just for documentation purposes, documenting the interface that it expects you to implement, there's no actual need for the `@implementer` and [Pyramid's own examples](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/security.html#writing-a-security-policy) don't use it. You won't find `ISecurityPolicy` formally implemented anywhere in the docs.